libpam/pam_dispatch.c

```diff
@@ -424,7 +424,6 @@ int _pam_dispatch(pam_handle_t *pamh, int flags, int choice)
     /* call the list of module functions */
     pamh->choice = choice;
     retval = _pam_dispatch_aux(pamh, flags, h, resumed, use_cached_chain);
-    resumed = PAM_FALSE;
```

Comment:
This might be a relic of defensive programming style when variables that are no longer expected to be used are assigned some safe defaults.  For example, ```close(fd), fd = -1```.
The best way to achieve this goal is not assigning safe defaults, though, but limiting the scope of variables, shifting the burden to the compiler.

Reply:
This sounds like some defensive programming, but it makes no sense to keep it. I've checked and it also doesn't make any sense to limit the scope of the variable, since this would mean the creation of a new scope just to limit it. I still think that the best approach is to remote the statement.